pkg/execution/executor/executor.go

```diff
@@ -887,6 +887,48 @@ func (e *executor) HandleResponse(ctx context.Context, i *runInstance) error {
 		"workflow_id", i.md.ID.FunctionID.String(),
 	)
 
+	// This is a success, which means either a generator or a function result.
+	if i.resp.Err == nil && len(i.resp.Generator) > 0 {
```

Comment:
A method for these `len(i.resp.Generator) > 0` checks would be awesome.

I remember there's some quirk here with an empty generator response also being represented by a single element with an `OpcodeNone`. 🤔

Reply:
Like this?
```go
// IsFunctionResult returns true if the response is a function result. It will
// return false if it's a step result.
func (r *DriverResponse) IsFunctionResult() bool {
	for _, op := range r.Generator {
		if op.Op != enums.OpcodeNone {
			return true
		}
	}
	return false
}
```